src/scripts/generateBridgeList.ts

```diff
@@ -0,0 +1,160 @@
+    )
+    .then((results) => {
+      return results.reduce((acc: TokenInfo[], res: any, index: number) => {
+        const token = calls[index]?.token
+
+        // Address of the token in the mapped chain
+        const address = '0x' + res[1].slice(-40)
+
+        // Get rid of tokens that are not known by the bridge contract
+        if (!token || address === ZERO_ADDRESS) return acc
+
+        const toReplace = tokensToReplace[token.address.toLowerCase()]
+        if (toReplace !== undefined) {
+          if (toReplace === null) {
+            // Do not add to the list
+            return acc
+          }
+
+          // Replace the fetched address with the one provided
+          acc.push({
+            ...token,
+            chainId,
+            address: toReplace,
+            extensions: undefined,
+          })
+
+          return acc
+        }
+        // Copy everything from the Mainnet token and override address and chainId
+        // Let's hope no one deploys the same token in a different networks with different data
+
+        const extensions = token.extensions as Extensions
+        const tokenAddress = extensions?.bridgeInfo?.[chainId]?.tokenAddress
+
+        if (tokenAddress && tokenAddress.toLowerCase() !== address.toLowerCase()) {
+          console.log(
+            `TokenList extension contains address different than returned by bridge. Extension: ${tokenAddress} Bridge: ${address}`,
+          )
+
+          acc.push({
+            ...token,
+            chainId,
+            address: tokenAddress,
+            extensions: undefined,
+          })
+
+          return acc
+        }
+
+        // Use bridge address
+        acc.push({
+          ...token,
+          chainId,
+          address,
+          extensions: undefined,
+        })
+
+        return acc
+      }, [])
+    })
+
+  writeTokenListToSrc(outputFilePath, { ...tokensList, tokens })
+
+  console.log(`${outputFilePath} is updated, tokens count: ${tokens.length}`)
+}
+
+export async function generateGnosisChainList(source: string | TokenList, resultFile: string) {
+  console.log('*** Map tokens from Mainnet to Gnosis chain using Omnibridge ***')
+
+  generateBridgedList(
+    SupportedChainId.GNOSIS_CHAIN,
+    source,
+    OMNIBRIDGE_ADDRESS,
+    OMNIBRIDGE_CONTRACT_ABI,
+    'bridgedTokenAddress',
+    resultFile,
+  )
+}
+
+export async function generateArbitrumOneChainList(source: string | TokenList, resultFile: string) {
+  console.log('*** Map tokens from Mainnet to Arbitrum One using Arbitrum bridge ***')
+
+  generateBridgedList(
+    SupportedChainId.ARBITRUM_ONE,
+    source,
+    ARBITRUM_BRIDGE_ADDRESS,
+    ARBITRUM_BRIDGE_ABI,
+    'calculateL2TokenAddress',
+    resultFile,
+    {
+      // USDC
+      '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48': '0xaf88d065e77c8cc2239327c5edb3a432268e5831',
+      // WETH
+      '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2': '0x82af49447d8a07e3bd95bd0d56f35241523fbab1',
+    },
+  )
```

Comment:
Manual overwrites, for known tokens for which we shouldn't use the bridged info.

Reply:
Would be great to move them into some json file

Reply:
Why only USDC and WETH?

Reply:
Those have their native versions on arb1

- WETH is the wrapped native ETH
- USDC has a native deployment as well

There might be others.
Actually, there are others
![image](https://github.com/cowprotocol/token-lists/assets/43217/15b2535a-8cbe-4775-98bf-07f72ca0d508)

Should I hard code them all?

Reply:
Wow, I didn't know. I'm ok with the current state

Reply:
I added them all